src/topp/MultiplexResolver.cpp

```diff
@@ -0,0 +1,674 @@
+    label_mass_shift_.insert(make_pair("ICPL6", getParam_().getValue("labels:ICPL6")));
+    label_mass_shift_.insert(make_pair("ICPL10", getParam_().getValue("labels:ICPL10")));
+  }
+
+  /**
+   * @brief split labels string
+   *
+   * @return list of samples containing lists of corresponding labels
+   */
+  std::vector<std::vector<String> > splitLabelString_()
+  {
+    std::vector<std::vector<String> > samples_labels;
+    std::vector<String> temp_samples;
+    
+    String labels(labels_);
+    boost::replace_all(labels, "[]", "no_label");
+    boost::replace_all(labels, "()", "no_label");
+    boost::replace_all(labels, "{}", "no_label");
+    boost::split(temp_samples, labels, boost::is_any_of("[](){}")); // any bracket allowed to separate samples
+    
+    for (unsigned i = 0; i < temp_samples.size(); ++i)
+    {
+      if (!temp_samples[i].empty())
+      {
+        if (temp_samples[i]=="no_label")
+        {
+          vector<String> temp_labels;
+          temp_labels.push_back("no_label");
+          samples_labels.push_back(temp_labels);
+        }
+        else
+        {
+          vector<String> temp_labels;
+          boost::split(temp_labels, temp_samples[i], boost::is_any_of(",;: ")); // various separators allowed to separate labels
+          samples_labels.push_back(temp_labels);
+        }
+      }
+    }
+    
+    if (samples_labels.empty())
+    {
+      vector<String> temp_labels;
+      temp_labels.push_back("no_label");
+      samples_labels.push_back(temp_labels);
+    }
+
+    return samples_labels;
+  }
+
+  /**
+   * @brief returns the relative delta mass between the first feature
+   * and the feature with the map index idx
+   *
+   * @param feature_handles    feature handles of a consensus feature
+   * @param idx    map index of interest
+   */
+  double DeltaMassFromMapIndex(const ConsensusFeature::HandleSetType& feature_handles, unsigned idx)
+  {
+    double first_mass = feature_handles.begin()->getMZ() * feature_handles.begin()->getCharge();
+    
+    for (ConsensusFeature::HandleSetType::const_iterator it_feat = feature_handles.begin(); it_feat != feature_handles.end(); ++it_feat)
+    {
+      if (it_feat->getMapIndex() == idx)
+      {
+        return it_feat->getMZ() * it_feat->getCharge() - first_mass;
+      }
+    }
+    
+    // return NaN if no matching index was found
+    return numeric_limits<double>::quiet_NaN();
+  }
+  
+  /**
+   * @brief check whether the theoretical delta mass pattern
+   * contains the label set of the detected pattern
+   *
+   * @param pattern    theoretical pattern
+   * @param label_set    label set of the detected pettern
+   * @param index_label_set    index within the pattern at which the label sets were matched
+   * 
+   * @return mass shift in the theoretical pattern where both label sets match
+   */
+  double matchLabelSet(const std::vector<MultiplexDeltaMasses::DeltaMass>& pattern, const MultiplexDeltaMasses::LabelSet& label_set, int& index_label_set)
```

Comment:
mixing function return values and by reference return values are considered bad style by someone on the internet. Consider returning a pair instead